### DISCUSSION TITLE
How to setup tsx for React Email?

### DISCUSSION THREAD

**Discussion by vicary** [floydspace/serverless-esbuild#516] [Q&A] [answered]
We would like to render React Email in lambda like advertised `render(<Email />)`.

Using the default config, the final bundle fails to transpile JSX syntax into `React.createElement`.

We tried adding the following config in our `serverless.yml` but it doesn't help.

```diff
custom:
  esbuild:
+    jsx: automatic
```

Is there anyway to make this happen?

**Comment by floydspace** [comment]
hi @vicary. According to [this esbuild document](https://esbuild.github.io/getting-started/#your-first-bundle) it must automatically bundle JSX without extra configuration. However, this plugin when bundling sets the `node` platform by default, as it is the most obvious target to bundle to for lambda functions, and I bet JSX is not supposed to be used in nodejs. what I would suggest is try [another platform option](https://esbuild.github.io/api/#platform). `neutral` or `browser` (does not make sense to me for lambda function though).
[Here](https://github.com/floydspace/serverless-esbuild/blob/fb4171977e400a3da0a0e54455a2566c57313fdc/src/index.ts#L320) you can find what is the plugin default option, among others.

**Reply by vicary** [reply, marked as answer]
Hey, thanks for the reply.

It turns out to be caused by esbuild node-resolve plugin forcing yarn's resolving to esbuild 0.17.

Installing the latest version of esbuild fixes the issue.
